src/Stratis.Bitcoin/Builder/FullNodeBuilder.cs

```diff
@@ -201,6 +202,33 @@ private IServiceCollection BuildServices()
             foreach (var configureFeature in this.featuresRegistrationDelegates)
                 configureFeature(this.Features);
 
+            // unregister features that depend on not registered features
+            while (true)
```

Comment:
I don't think you need this  loop (especially not a while(true) ) just foreach the feature and call `EnsureDependencies` we should not continue if any of the dependencies is broken registering is not needed.